crates/starknet-devnet-server/src/api/http/endpoints/dump_load.rs

```diff
@@ -1,50 +1,37 @@
 use axum::extract::State;
 use axum::Json;
 
+use super::extract_optional_json_from_request;
 use crate::api::http::error::HttpApiError;
 use crate::api::http::models::{DumpPath, LoadPath};
 use crate::api::http::{HttpApiHandler, HttpApiResult};
 use crate::api::Api;
 
 pub async fn dump(
     State(state): State<HttpApiHandler>,
-    Json(path): Json<DumpPath>,
+    optional_path: Option<Json<DumpPath>>,
 ) -> HttpApiResult<()> {
-    dump_impl(&state.api, path).await
+    dump_impl(&state.api, extract_optional_json_from_request(optional_path)).await
 }
 
-pub(crate) async fn dump_impl(api: &Api, path: DumpPath) -> HttpApiResult<()> {
+pub(crate) async fn dump_impl(api: &Api, path: Option<DumpPath>) -> HttpApiResult<()> {
     let starknet = api.starknet.write().await;
 
     if starknet.config.dump_on.is_none() {
         return Err(HttpApiError::DumpError {
             msg: "Please provide --dump-on mode on startup.".to_string(),
         });
     }
-
-    match path.path {
-        None => {
-            // path not present
-            starknet
-                .dump_events()
-                .map_err(|err| HttpApiError::DumpError { msg: err.to_string() })?;
-            Ok(())
-        }
-        Some(path) => {
-            if !path.is_empty() {
-                // path is present and it's not empty
-                starknet
-                    .dump_events_custom_path(Some(path))
-                    .map_err(|err| HttpApiError::DumpError { msg: err.to_string() })?;
-                Ok(())
-            } else {
-                // path is present but it's empty
-                starknet
-                    .dump_events()
-                    .map_err(|err| HttpApiError::DumpError { msg: err.to_string() })?;
-                Ok(())
-            }
-        }
+    let path = path.map_or(String::new(), |s| s.path.clone());
+    // path not present
+    if path.is_empty() {
+        starknet.dump_events().map_err(|err| HttpApiError::DumpError { msg: err.to_string() })?;
+        Ok(())
+    } else {
+        starknet
+            .dump_events_custom_path(Some(path))
+            .map_err(|err| HttpApiError::DumpError { msg: err.to_string() })?;
+        Ok(())
```

Comment:
Great that someone has finally refactored this function. `Ok(())` can be extracted.

Reply:
17e47a98d84155d984f03c4f772685d71ef88c3d